frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditNumber.tsx

```diff
@@ -0,0 +1,193 @@
+    handleCancel,
+    setValue,
+  } = useEditFieldForm<EditNumberInputs, NumberFieldBase>({
+    field,
+    transform: {
+      input: transformNumberFieldToEditForm,
+      output: transformNumberEditFormToField,
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const watchedSelectedValidation = watch(
+    'ValidationOptions.selectedValidation',
+  )
+
+  const customValValidationOptions: RegisterOptions = useMemo(
+    () => ({
+      required: {
+        value: watchedSelectedValidation !== '',
+        message: 'Please enter number of characters',
+      },
+      min: {
+        value: 1,
+        message: 'Cannot be less than 1',
+      },
+      max: {
+        value: 10000,
+        message: 'Cannot be more than 10000',
+      },
+    }),
+    [watchedSelectedValidation],
+  )
+
+  // Effect to clear validation option errors when selection limit is toggled off.
+  useEffect(() => {
+    if (watchedSelectedValidation === '') {
+      clearErrors('ValidationOptions')
+      setValue('ValidationOptions.customVal', '')
+    }
+  }, [clearErrors, setValue, watchedSelectedValidation])
+
+  return (
+    <DrawerContentContainer>
+      <FormControl isRequired isReadOnly={isLoading} isInvalid={!!errors.title}>
+        <FormLabel>Question</FormLabel>
+        <Input autoFocus {...register('title', requiredValidationRule)} />
+        <FormErrorMessage>{errors?.title?.message}</FormErrorMessage>
+      </FormControl>
+      <FormControl
+        isRequired
+        isReadOnly={isLoading}
+        isInvalid={!!errors.description}
+      >
+        <FormLabel>Description</FormLabel>
+        <Textarea {...register('description')} />
+        <FormErrorMessage>{errors?.description?.message}</FormErrorMessage>
+      </FormControl>
+      <FormControl isReadOnly={isLoading}>
+        <Toggle {...register('required')} label="Required" />
+      </FormControl>
+      <FormControl
+        isReadOnly={isLoading}
+        isInvalid={!isEmpty(errors.ValidationOptions)}
+      >
+        <FormLabel isRequired>Number of characters allowed</FormLabel>
+        <SimpleGrid mt="0.5rem" columns={2} spacing="0.5rem">
+          <Controller
+            name="ValidationOptions.selectedValidation"
+            control={control}
+            render={({ field }) => (
+              <SingleSelect
+                items={Object.values(NumberSelectedValidation)}
+                {...field}
+              />
+            )}
+          />
+          <Controller
+            name="ValidationOptions.customVal"
+            control={control}
+            rules={customValValidationOptions}
+            render={({ field }) => (
+              <NumberInput
+                flex={1}
+                showSteppers={false}
+                {...field}
+                placeholder="Number of characters"
+                isDisabled={watchedSelectedValidation === ''}
+              />
```

Comment:
will `NumberInput` also allow other numbers without like `e`? But I guess validation will filter it out due to min and max